components/browser/state/src/main/java/mozilla/components/browser/state/search/SearchEngine.kt

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.search
+
+import android.graphics.Bitmap
+
+/**
+ * A data class representing a search engine.
+ *
+ * @property id the ID of this search engine.
+ * @property name the name of this search engine.
+ * @property icon the icon of this search engine.
+ * @property type the type of this search engine.
+ * @property resultUrls the list of the queried suggestions result urls.
+ * @property suggestUrl the search suggestion url.
+ */
+data class SearchEngine(
+    val id: String,
+    val name: String,
+    val icon: Bitmap? = null,
```

Comment:
Seeing that `Bitmap` in `SearchEngine` in `browser-search` is not nullable. Maybe we can get away with not making it nullable here too? 🤔 But not super important right now, I think.

Reply:
I'll make it not nullable.